### PR TITLE
Don't save Draft if there is no modification

### PR DESCRIPTION
**Pull request by KevinBoulongne**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt [hunk 1]
```diff
@@ -132,6 +138,18 @@ class NewMessageViewModel(application: Application) : AndroidViewModel(applicati
...
     }
 
+    private fun saveDraftSnapshot() {
+        snapshot = DraftSnapshot(
+            mailTo.toSet(),
+            mailCc.toSet(),
+            mailBcc.toSet(),
+            mailSubject,
+            mailBody,
+            // mailSignature,
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt:148]: Remove useless comments

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt [hunk 2]
```diff
@@ -209,6 +230,14 @@ class NewMessageViewModel(application: Application) : AndroidViewModel(applicati
         }
     }
 
+    private fun checkDraftSnapshot() {
+        if (snapshot.isTheSame()) {
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt:234]: Rename to `hasNoChanges` instead

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt [hunk 3]
```diff
@@ -260,6 +289,26 @@ class NewMessageViewModel(application: Application) : AndroidViewModel(applicati
...
+        val attachmentsUuids: Set<String>,
+    )
+
+    private fun DraftSnapshot.isTheSame(): Boolean {
+        return to == mailTo.toSet() &&
+                cc == mailCc.toSet() &&
+                bcc == mailBcc.toSet() &&
+                subject == mailSubject &&
+                body == mailBody &&
+                // signature == mailSignature &&
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt]: useless

**Review by sirambd — CHANGES_REQUESTED** [review]
This feature need logical changes.

**Review by sirambd — CHANGES_REQUESTED** [review]

**Review by sirambd — APPROVED** [review]

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**